File: water_container_max.java
class Solution {
    public int maxArea(int[] height) {
        
        int i=0, j=height.length-1;
        int max=0;
        while(i!=j){
            
            System.out.println();
            if(max< Math.min(height[i],height[j])*(j-1)){
                max=Math.min(height[i],height[j])*(j-1);
            }
            if(height[i]<height[j]){
                i++;
            }else if(height[i]>height[j]){
                j--;
            }
            
            
        }
        return max;
    }
}

File: Water_Container_Max.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max1=0,low=0, high= height.size()-1;
        while(low<high){
            int currmax = min(height[low],height[high]) * (high-low);
            max1 = max(max1, currmax);
            if(height[low]<height[high]){
                low++;
            }
            else{
                high--;
            }
        }
        return max1;
        
    }
};
